app/src/main/java/com/example/mob_lab1/MainActivity.java

```diff
@@ -0,0 +1,93 @@
+
+    public int passMinLenght = 8;
+    public String emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mFirebaseAuth = FirebaseAuth.getInstance();
+        emailID = findViewById(R.id.editText);
+        password = findViewById(R.id.editText3);
+        btnSignUp = findViewById(R.id.button);
+        tvSignIn = findViewById(R.id.textView);
+        btnSignUp.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                String email = emailID.getText().toString();
+                String pwd = password.getText().toString();
+
+                if(email.isEmpty()){
+                    emailID.setError("Please enter your email");
+                    emailID.requestFocus();
+                }
+                else if(pwd.isEmpty()){
+                    password.setError("Please enter your password");
+                    password.requestFocus();
+                }
+                else if(email.isEmpty() && pwd.isEmpty()){
+                    Toast.makeText(MainActivity.this, "Fields are empty!",Toast.LENGTH_SHORT).show();
+                }
+                else if(password.length() < passMinLenght){  // Validation of password
+                    password.setText("");
+                    password.setError("Minimal lengt of password should be 8!");
+                }
+                else if(!emailID.getText().toString().trim().matches(emailPattern)){   // Validation of email
+                    emailID.setText("");
+                    emailID.setError("Please re-enter your email correctly");
+                }
+                else if(!(email.isEmpty() && pwd.isEmpty())){
+                    mFirebaseAuth.createUserWithEmailAndPassword(email, pwd).addOnCompleteListener(MainActivity.this, new OnCompleteListener<AuthResult>() {
+                        @Override
+                        public void onComplete(@NonNull Task<AuthResult> task) {
+                            if(!task.isSuccessful()){
+                                Toast.makeText(MainActivity.this, "Sign Up Unsuccesful. Please try again.",Toast.LENGTH_SHORT).show();
+                            }
+                            else {
+                                startActivity(new Intent(MainActivity.this, HomeActivity.class ));
+                            }
+                        }
+                    });
+                }
+                else{
+                    Toast.makeText(MainActivity.this, "Error Occured!",Toast.LENGTH_SHORT).show();
+                }
+            }
+        });
+
+        tvSignIn.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent i = new Intent(MainActivity.this, LoginActivity.class);

```

Comment:
`i` is incorrect name
```suggestion
                Intent intent = new Intent(MainActivity.this, LoginActivity.class);
```